graphrag/api/prompt_tune.py

```
@@ -111,7 +111,7 @@ async def generate_indexing_prompts(
 
     # if max_retries is not set, inject a dynamically assigned value based on the number of expected LLM calls
     # to be made or fallback to a default value in the worst case
-    if default_llm_settings.max_retries == -1:
+    if default_llm_settings.max_retries < -1:
```

-1 means the max retries will be dynamically calculated.
When would it be less than -1?

---

It should never be less than one. This was just some logic to address an edge case where the state is "undefined"; like a typo for example. 

Should we raise an exception when the value of `max_retries` is out of bounds? Or should we assume that any value lower than 0 means calculate `max_retries` dynamically?

---

I would like that we iterate over this. This should be handed and resolved when the config is loaded.

Approving since this change is required for experimentation

---

Agree.